pkg/apis/apps/validation/validation.go

```diff
@@ -155,12 +173,16 @@ func ValidateStatefulSetUpdate(statefulSet, oldStatefulSet *apps.StatefulSet) fi
 	restoreStrategy := statefulSet.Spec.UpdateStrategy
 	statefulSet.Spec.UpdateStrategy = oldStatefulSet.Spec.UpdateStrategy
 
+	deletePolicy := statefulSet.Spec.PersistentVolumeClaimDeletePolicy
+	statefulSet.Spec.PersistentVolumeClaimDeletePolicy = oldStatefulSet.Spec.PersistentVolumeClaimDeletePolicy
+
 	if !apiequality.Semantic.DeepEqual(statefulSet.Spec, oldStatefulSet.Spec) {
-		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden"))
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec"), "updates to statefulset spec for fields other than 'replicas', 'template', 'updateStrategy' and 'persistentVolumeClaimDeletePolicy' are forbidden"))
 	}
 	statefulSet.Spec.Replicas = restoreReplicas
 	statefulSet.Spec.Template = restoreTemplate
 	statefulSet.Spec.UpdateStrategy = restoreStrategy
+	statefulSet.Spec.PersistentVolumeClaimDeletePolicy = deletePolicy
```

Comment:
The new deletePolicy also needs to be validated

Reply:
Thx for clarifying this offline. Done.